### PR TITLE
fix #1827 ComponentScan with SPEL will not parse

### PR DESCRIPTION
**Pull request by qing-wq**
Fixes https://github.com/langchain4j/langchain4j/issues/1827

fix ComponentScan basePackages with placeholders will not be parsed. 

Changes Made:
- Support for parsing @ComponentScan with placeholders
- Simplify the getBasePackages()

### CODE REVIEW

**Comment by qing-wq** [comment]
I added new test cases and removed the redundant tests about SpringBootApplication, which was not required

**Comment by langchain4j** [comment]
@qing-wq thanks! Why removing existing tests though?

**Comment by qing-wq** [comment]
I've learned that Spring Bootapplication is essentially an alias for ComponentScan, so I don't think you need to test it alone, just@ComponentScan：
```java
@AliasFor(annotation = ComponentScan.class, attribute = "basePackages")
String[] scanBasePackages() default {};

@AliasFor(annotation = ComponentScan.class, attribute = "basePackageClasses")
Class<?>[] scanBasePackageClasses() default {};
```

**Comment by langchain4j** [comment]
@qing-wq I would still keep it, assuming there is no harm and test is fast

**Comment by qing-wq** [comment]
I understand, and I canceled my modification

**Comment by langchain4j** [comment]
@qing-wq please check the tests, they all failed.

**Comment by qing-wq** [comment]
sorry, there are some errors in my test code, but now I have fixed it.

File: langchain4j-spring-boot-starter/src/test/java/dev/langchain4j/service/spring/mode/automatic/differentPackage/package1/DifferentPackageAiServiceIT.java [hunk 1]
```diff
@@ -98,4 +102,29 @@ void should_create_AI_service_that_use_componentScan_basePackageClasses() {
...
+
+        contextRunner
+                .withPropertyValues(
+                        "langchain4j.open-ai.chat-model.api-key=" + OPENAI_API_KEY,
+                        "langchain4j.open-ai.chat-model.max-tokens=20",
+                        "langchain4j.open-ai.chat-model.temperature=0.0",
+                        "test.basePackages=dev.langchain4j.service.spring.mode.automatic.differentPackage.package2"
+                )
+                .withUserConfiguration(DifferentPackageAiServiceApplication.class)
+                .withUserConfiguration(ComponentScanWithBasePackageClasses.class)
```
Comment (langchain4j) [inline langchain4j-spring-boot-starter/src/test/java/dev/langchain4j/service/spring/mode/automatic/differentPackage/package1/DifferentPackageAiServiceIT.java:117]:
```suggestion
                .withUserConfiguration(ComponentScanWithPlaceholder.class)
```

**Review by langchain4j — APPROVED** [review]
@qing-wq thank you!

**Pull request merged**